pkg/controllers/multicluster/serviceexport_controller.go

```diff
@@ -69,8 +72,32 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	// Mark ServiceExport to be deleted, which is indicated by the deletion timestamp being set.
 	isServiceExportMarkedForDelete := serviceExport.GetDeletionTimestamp() != nil
 
+	clusterPropertyForClusterId := &aboutv1alpha1.ClusterProperty{}
+	clusterPropertyForClustersetId := &aboutv1alpha1.ClusterProperty{}
+	if !isServiceExportMarkedForDelete {
+		// Fetch ClusterProperty with name "id.k8s.io"
+		err := r.Client.Get(ctx, client.ObjectKey{Name: ClusterIdName}, clusterPropertyForClusterId)
+		if err != nil {
+			r.Log.Error(err, "error fetching ClusterProperty with name "+ClusterIdName)
+			return ctrl.Result{}, err
+		} else {
+			r.Log.Debug("ClusterProperty with ClusterId found", "ClusterId", clusterPropertyForClusterId.Spec.Value)
+		}
+
+		// Fetch ClusterProperty with name "clusterset.k8s.io"
+		err = r.Client.Get(ctx, client.ObjectKey{Name: ClustersetIdName}, clusterPropertyForClustersetId)
```

Comment:
Instead of making two API calls you can fetch ClusterProperty list, and iterate over them to verify both clustersetId and clusterId objects. Also, consider moving this duplicated logic to utils as this is used in both the controllers.